test/e2e/apps/statefulset.go

```diff
@@ -1156,6 +1157,45 @@ var _ = SIGDescribe("StatefulSet", func() {
 				return false, nil
 			})
 		})
+
+		/*
+			Release: v1.32
+			Testname: StatefulSet, PodIndexLabel
+			Description: When Pods for a StatefulSet is created it MUST be created with PodIndexLabel feature
+			Attempt to read the PodIndexLabel key and value
+		*/
+		framework.ConformanceIt("pods for Statefulset should have PodIndexLabel with right index", feature.PodIndexLabel, func(ctx context.Context) {
+			one := int64(1)
+			ssName := "test-ss"
+
+			// Define StatefulSet Labels
+			ssPodLabels := map[string]string{
+				"name": "sample-pod",
+				"pod":  WebserverImageName,
+			}
+			ss := e2estatefulset.NewStatefulSet(ssName, ns, headlessSvcName, 3, nil, nil, ssPodLabels)
+			setHTTPProbe(ss)
+			ss, err := c.AppsV1().StatefulSets(ns).Create(ctx, ss, metav1.CreateOptions{})
+			framework.ExpectNoError(err)
+			e2estatefulset.WaitForRunningAndReady(ctx, c, *ss.Spec.Replicas, ss)
+			waitForStatus(ctx, c, ss)
+
+			ginkgo.By("checking the index label and value of all pods")
+			pods := e2estatefulset.GetPodList(ctx, c, ss)
+			labelIndices := sets.NewInt()
+			for _, pod := range pods.Items {
+				ix, err := strconv.Atoi(pod.Labels[appsv1.PodIndexLabel])
+				framework.ExpectNoError(err, "failed obtaining label with key: %s from pod: %s in namespace: %s", appsv1.PodIndexLabel, pod.Name, pod.Namespace)
+				labelIndices.Insert(ix)
+			}
+			wantIndexes := []int{0, 1, 2}
+			gotIndexes := labelIndices.List()
+			gomega.Expect(gotIndexes).To(gomega.Equal(wantIndexes), "expected indexes for label with key %s: %s, but got %s", appsv1.PodIndexLabel, wantIndexes, gotIndexes)
```

Comment:
as for Job, please check if the want and got is already output by gomega, if it is, then not need to repeat. Instead output the sts name and namespace.

Reply:
updated